### PR TITLE
CU-2t2ew2w - update funding tests with integrations

### PR DESCRIPTION
**Pull request by 0xangelo**
## Issue
[CU-2t2ew2w](https://app.clickup.com/t/2t2ew2w)

## Description
Also includes a fix to the problem of outstanding profits after closing a position (linked in the ClickUp issue above.) This is a temporary solution until further testing ensures its safety.

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2t2ew2w Integration tests with Vamm Pallet](https://app.clickup.com/t/2t2ew2w)

File: frame/clearing-house/src/integrations/tests.rs [hunk 1]
```diff
@@ -118,155 +145,454 @@ impl Default for MarketConfig {
...
-			taker_fee: 0,
-			twap_period: ONE_HOUR,
-		};
-		assert_ok!(TestPallet::create_market(Origin::signed(ALICE), config));
 
-		assert_ok!(TestPallet::deposit_collateral(Origin::signed(BOB), USDC, UNIT * 100));
+	#[test]
+	#[ignore = "FIXME: vAMM TWAP isn't updated if last twap timestamp is equal to the current \
+	block's timestamp"]
+	fn should_update_vamm_twap_in_the_same_block() {
```
Comment (0xangelo) [inline frame/clearing-house/src/integrations/tests.rs:330]: @Cardosaum this test could be useful for you when evaluating how to update the vAMM's TWAP for intra-block trades

File: frame/clearing-house/src/lib.rs [hunk 2]
```diff
@@ -1111,11 +1120,22 @@ pub mod pallet {
...
+				let market = Self::try_get_market(&position.market_id)?;
+				Self::settle_funding(position, &market, &mut collateral)?;
+			}
+
+			let mut available_profits = Self::available_profits().unwrap_or_else(Zero::zero);
+			if !available_profits.is_zero() {
+				let mut outstanding_profits =
+					Self::outstanding_profits(account_id).unwrap_or_else(Zero::zero);
+
+				let realizable_profits = cmp::min(outstanding_profits, available_profits);
```
Comment (Cardosaum) [inline frame/clearing-house/src/lib.rs:1132]: Any reason not to use `outstanding_profits.min(available_profits)` over `cmp::min(outstanding_profits, available_profits)`?

I see you're using the `min` method from the `sp_std` crate, but wouldn't this method be used in either syntax anyway?

**Review by Cardosaum — DISMISSED** [review]
LGTM

:ship:

File: frame/clearing-house/src/lib.rs (hunk 2 above)
Comment (0xangelo) [reply frame/clearing-house/src/lib.rs:1132]: I think I just forgot to update that part to use the trait method. Thanks for pointing it out!

**Review by Cardosaum — APPROVED** [review]

**Review by kevin-composable — APPROVED** [review]
VAMM integration tests are looking great

**Comment by 0xangelo** [comment]
@Mergifyio rebase

**Pull request closed without merging**